generator/internal/mixin_utils_test.cc

```diff
@@ -26,8 +26,41 @@ using ::google::protobuf::FileDescriptor;
 using ::google::protobuf::ServiceDescriptor;
 using ::testing::AllOf;
 using ::testing::Contains;
+using ::testing::ElementsAre;
+using ::testing::Eq;
+using ::testing::Field;
 using ::testing::NotNull;
 
+bool is_http_rule_expected(google::api::HttpRule const& http_rule,
+                           std::string const& body, std::string const& path,
+                           google::api::HttpRule::PatternCase pattern_case) {
+  if (http_rule.body() != body) return false;
```

Comment:
If we went for this approach, we would ideally log which fields were mismatched. It'd probably be better to write your own matcher using `::testing::AllOf(::testing::ResultOf(...), ::testing::ResultOf(...))` to match the individual fields.

But I don't think we should do either. I think we should just verify the `HttpRule` proto is what we expect by parsing a proto string, and checking `google::cloud::testing_util::IsProtoEqual(expected)`.

Reply:
Replaced by IsProtoEqual.